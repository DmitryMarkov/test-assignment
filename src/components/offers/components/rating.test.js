import React from 'react'
import { render } from '@testing-library/react'
import Rating, { calculateRating } from './rating'

test('Rating should render properly', () => {
  const mockRating = 90
  const mockVotesCount = 5
  const { asFragment, getByText } = render(
    <Rating rating={mockRating} votesCount={mockVotesCount} />
  )

  expect(getByText(/Rating/)).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

test('calculateRating works correctly', () => {
  const mockRating1 = 90
  const mockRating2 = 0
  expect(calculateRating(mockRating1)).toBe('4.5')
  expect(calculateRating(mockRating2)).toBe('0.0')
})
