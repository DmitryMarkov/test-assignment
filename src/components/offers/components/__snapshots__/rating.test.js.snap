// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render properly 1`] = `
<DocumentFragment>
  <div
    class="css-dmeso1"
  >
    Rating: 4.5 (5 votes)
  </div>
</DocumentFragment>
`;
